### PR TITLE
GeniusReader loader

### PR DESCRIPTION
**Pull request by ThibaudARoy**
# Description

Added GeniusReader loader, which allows for loading of documents from the Genius API

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [X] New Loader/Tool

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have added a library.json file if a new loader/tool was added
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by nerdai** [comment]
Thanks for the contribution @ThibaudARoy. Looks like test our failing due to import errors. Can you follow a similar pattern to whats been used in the past, which is to do lazy imports and throw an  error if the module is not installed?

See here: https://github.com/run-llama/llama-hub/blob/364feb5b9c41a081da53aae02e894c6caa2f478d/llama_hub/llama_packs/gradio_agent_chat/base.py#L36

File: llama_hub/genius/base.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+"""Genius Reader."""
+from typing import List, Optional
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+import lyricsgenius
```
Comment (nerdai) [inline llama_hub/genius/base.py:5]: please see my comment in main conversation thread. Can we move this into the __init__ of the class?

**Review by nerdai — CHANGES_REQUESTED** [review]
please move `lyricsgenious` import statement

**Comment by jerryjliu** [comment]
@ThibaudARoy is there an update on this?

**Comment by ThibaudARoy** [comment]
@jerryjliu updated the import statement. Let me know if there's anything else I can fix.

**Comment by jerryjliu** [comment]
sorry for delay @ThibaudARoy seems like there's some minor linting errors you can fix with `make lint; make format`!

**Pull request merged**